modules/http-helper/continuous.go

```diff
@@ -0,0 +1,29 @@
+package http_helper
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+)
+
+// Continuously check the given URL every 1 second until the stopChecking channel receives a signal to stop.
+func ContinuouslyCheckUrl(t *testing.T, url string, stopChecking <-chan bool, sleepBetweenChecks time.Duration) {
+	go func() {
+		for {
+			select {
+			case <-stopChecking:
+				logger.Logf(t, "Got signal to stop downtime checks for URL %s.\n", url)
+				return
+			case <-time.After(sleepBetweenChecks):
+				statusCode, body, err := HttpGetE(t, url)
+				logger.Logf(t, "Got response %d and err %v from URL at %s", statusCode, err, url)
+				if err != nil {
+					t.Fatalf("Failed to make HTTP request to the URL at %s: %s\n", url, err.Error())
```

Comment:
Yes this works. It's been causing my tests to fail (and me pulling hair) the past 2 business days 😄 Also, I copy pasted this with no modifications out of `module-ecs`.

Reply:
Hmm...

https://github.com/ipfs/go-ipfs/issues/2043
https://github.com/golang/go/issues/24678
https://github.com/golang/go/issues/15758

Reply:
Ah ok I see. So the bug is that there are race conditions in there. Ok I'll see what I can do to fix this.

Reply:
Ah, nice find! You should add a comment that indicates why we specifically use `Errorf` to each spot in the code so we don't accidentally break that.

One other way to do this would be for the method to return a channel and to write to that channel an error if something went wrong or `nil` if everything finished successfully. The calling code could then read from that channel to see how things went. But `Errorf` is probably good enough for now that it's not worth adding this complexity.